src/js/api/order.js

```diff
@@ -549,6 +547,165 @@ ripe.Ripe.prototype.logOrderP = function(number, options) {
+ *
+ * @param {Number} number The number of the order to get the number of chat lines from.
+ * @param {Object} options An object of options to configure the request.
+ * @param {Function} callback Function with the result of the request.
+ * @returns {XMLHttpRequest} The XMLHttpRequest instance of the API request.
+ */
+ripe.Ripe.prototype.chatLinesCountOrder = function(number, options, callback) {
+    callback = typeof options === "function" ? options : callback;
+    options = typeof options === "function" || options === undefined ? {} : options;
+    const url = `${this.url}orders/${number}/chat/lines/count`;
+    options = Object.assign(options, {
+        url: url,
+        method: "GET",
+        auth: true
+    });
+    options = this._build(options);
+    return this._cacheURL(options.url, options, callback);
+};
+
+/**
+ * Returns the number of general chat lines of an order, unrelated to any state.
+ *
+ * @param {Number} number The number of the order to get the number of chat lines from.
+ * @param {Object} options An object of options to configure the request.
+ * @returns {Promise} The number of order chat's lines.
+ */
+ripe.Ripe.prototype.chatLinesCountOrderP = function(number, options) {
+    return new Promise((resolve, reject) => {
+        this.chatLinesCountOrder(number, options, (result, isValid, request) => {
+            isValid ? resolve(result) : reject(new ripe.RemoteError(request, null, result));
+        });
+    });
+};
+
+/**
+ * Adds a new general chat line message to a specific order.
+ *
+ * @param {Number} number The number of the order where the chat line message will be added.
+ * @param {Object} contents The contents of the chat message.
+ * @param {Object} options An object of options to configure the request.
+ * @param {Function} callback Function with the result of the request.
+ * @returns {XMLHttpRequest} The XMLHttpRequest instance of the API request.
+ */
+ripe.Ripe.prototype.chatCreateLineOrder = function(number, contents, options, callback) {
+    callback = typeof options === "function" ? options : callback;
+    options = typeof options === "function" || options === undefined ? {} : options;
+    const url = `${this.url}orders/${number}/chat/lines`;
+    options = Object.assign(options, {
+        url: url,
+        method: "POST",
+        params: {
+            contents: contents
+        },
+        auth: true
+    });
+    options = this._build(options);
+    return this._cacheURL(options.url, options, callback);
+};
+
+/**
+ * Adds a new general chat line message to a specific order.
+ *
+ * @param {Number} number The number of the order where the chat line message will be added.
+ * @param {Object} contents The contents of the chat message.
+ * @param {Object} options An object of options to configure the request.
+ * @returns {Promise} An object containing the chat lines for the specific order.
+ */
+ripe.Ripe.prototype.chatCreateLineOrderP = function(number, contents, options) {

```

Comment:
to follow the same format as other methods such as `createReturnWaybillOrderP`

```suggestion
ripe.Ripe.prototype.createChatLineOrderP = function(number, contents, options) {

```

Reply:
Following the same formatting as state chat: https://github.com/ripe-tech/ripe-sdk/blob/jc/4702-order-chat-methods/src/js/api/order.js#L914

Reply:
> Following the same formatting as state chat: https://github.com/ripe-tech/ripe-sdk/blob/jc/4702-order-chat-methods/src/js/api/order.js#L914

I understand that but I think we follow the same format that is in the majority of the methods to keep consistency

Reply:
But it should also be a direct naming from Core, which was already approved: https://github.com/ripe-tech/ripe-core/blob/master/src/ripe_core/controllers/api/order.py#L226

Reply:
Then I guess the effort to change isn't worth